File: fraud-detection-pipeline/src/foam.kt
package com.affirm.fraud

import com.foam.kotlin.Foam
import com.affirm.fraud.config.AppConfig

fun initializeFoamMonitoring(config: AppConfig) {
    Foam.init(
        serviceName = "fraud-detection-pipeline",
        isProduction = config.isProduction,
        apiKey = config.foamApiKey,
        environment = config.environment,
        tags = mapOf(
            "team" to "trust-and-safety",
            "domain" to "fraud-ml",
            "model-version" to config.modelVersion
        )
    )
}

File: merchant-integration-gateway/src/foam.py
import foam
from gateway.config import settings

foam.init(
    service_name="merchant-integration-gateway",
    is_production=settings.IS_PRODUCTION,
    api_key=settings.FOAM_API_KEY,
    environment=settings.ENVIRONMENT,
    tags={"team": "partner-engineering", "domain": "integrations"},
    sample_rate=0.1 if settings.IS_PRODUCTION else 1.0
)